test/window-test.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-env mocha */
+/* eslint prefer-arrow-callback: "off" */
+
+'use strict';
+
+const assert = require('bsert');
+const SlidingWindow = require('../lib/net/window');
+
+async function sleep(time) {
+  return new Promise(resolve => setTimeout(resolve, time));
+}
+
+describe('SlidingWindow (Unit)', function() {
+  const window = new SlidingWindow({
+    window: 10,
+    limit: 100
+  });
+
+  beforeEach(() => {
```

Comment:
Should these be `{before,after}Each` or just `before` and `after`? The tests pass with either. Calling `stop` will causes the `weight` calculation to be much higher when `score` is called

Reply:
With `beforeEach` and `afterEach` the windows start over from the start of the test case, so any timing effects outside the test case are nullified.